common/build/eslint-plugin-fluid/package.json

```diff
@@ -0,0 +1,59 @@
+{
+	"name": "@fluid-internal/eslint-plugin-fluid",
+	"version": "0.1.0",
+	"private": true,
+	"description": "Custom ESLint rules for the Fluid Framework",
+	"homepage": "https://fluidframework.com",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/microsoft/FluidFramework.git",
+		"directory": "common/build/eslint-pluig-fluid/"
+	},
+	"license": "MIT",
+	"author": "Microsoft and contributors",
+	"main": "index.js",
+	"scripts": {
+		"build": "npm run build:readme",
+		"build:readme": "markdown-magic --files \"**/*.md\"",
+		"clean": "rimraf --glob nyc",
+		"cleanup-printed-configs": "node ./scripts/cleanup-printed-configs.js ./printed-configs",
+		"format": "npm run prettier:fix",
+		"prettier": "prettier --check . --cache --ignore-path ../../../.prettierignore",
+		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
+		"test": "mocha src/test"
```

Comment:
Have we tested this script? I'm not sure mocha will be able to run `.ts` test files without further help, we usually point it to the compiled files in `dist/`.

Reply:
https://github.com/microsoft/FluidFramework/blob/22b05b09c0881a04633fae4712c794da77909411/common/build/eslint-config-fluid/package.json#L29

I'm using `eslint-config-fluid` as a reference, and this was how the tests were run in the package. Should we change both packages so that the `test` script points the `dist` folder?

Reply:
<img width="937" alt="Screenshot 2024-02-14 at 13 53 59" src="https://github.com/microsoft/FluidFramework/assets/111468570/a49f0f8e-dcdc-4682-9cd8-028a65ff3022">

With the changes from `"./mockModule.js";` to `"./mockModule.ts";` the tests are workings.

Reply:
Oh true, the test files in this case aren't even compiled (these packages don't have a "real" build), they're defined directly as `.test.js` files. The `test` script is probably "wider" than it should be, it should target `src/test/**/*.test.*js`, can you try that? I'm almost certain it's not looking at the `.ts` files for tests, those are just "helper files" that the tests use. It's processing all `.js` files under src/test looking for tests though. We only have the two which are indeed tests, but it'd be nice if the script was more targeted in what it cares about.

Reply:
> With the changes from "./mockModule.js"; to "./mockModule.ts"; the tests are workings.

Hmm... ok, the fact that they're not actually running, just being linted, leaves me wondering if using `.js` in the import would work or not. Did you try that option too? Do the tests fail in that case? If they pass, I say let's move to `.js`. And if we need `.ts`, let's add a comment specifying why in this particular case it's ok and deliberate (we're only passing these files to eslint for testing without any build transforming them into js/cjs/mjs).

Reply:
Changing to `.js` would not work for since we specify types in `exceptionFile.ts` and `mockModule.js`, and since these must stay as `.ts`, syntax below seems reasonable for me (except that the linter is complaining):
- `import { publicFunction, internalFunction, alphaFunction } from "./mockModule.ts";`
- `import { exceptionInternalFunction, alphaFunction2 } from "./exceptionFile.ts";`

<img width="955" alt="Screenshot 2024-02-14 at 14 31 37" src="https://github.com/microsoft/FluidFramework/assets/111468570/84ae6991-b770-4fb1-95fb-17a67226e630">

I'm guessing there could be a problem with the current `tsc` setting, thus the it's having problem finding `mockModule.js` although the path is specified as `.ts` 

https://github.com/microsoft/FluidFramework/blob/a175505e4ffbdbde2a2165e00da1e42a32985bd3/common/build/eslint-config-fluid/src/test/enforce-no-restricted-tags-imports/enforce-no-restricted-tags-imports.test.js#L69

https://github.com/microsoft/FluidFramework/blob/a175505e4ffbdbde2a2165e00da1e42a32985bd3/common/build/eslint-config-fluid/src/test/enforce-no-restricted-tags-imports/enforce-no-restricted-tags-imports.test.js#L96

Reply:
Well, actually it has a point that the `.js` versions of the files don't exist; we never give them to Typescript to compile from `.ts` to `.js`.

Let's keep `.ts` here but do add a note like I suggested in my previous comment, because it goes against conventions we use across the repo.